### PR TITLE
Set `is_pyi` if `stdin_filename` ends with `.pyi`

### LINKED ISSUE #2167: Determine file type from `--stdin-filename`
**Is your feature request related to a problem? Please describe.**

Currently, `black` ignores the filename passed to `--stdin-filename` to determine if it is a python file or stub file for formatting. This is frustrating for me because I have to set up a separate file type for python stub files to use `coc.nvim` + `coc-diagnostic` in order to use a different command line invocation (adding `--pyi`) to process stub files.

**Describe the solution you'd like**

When `black` is invoked with `--stdin-filename`, it should set the `is_pyi` flag on the `Mode` like what is done at the beginning of `format_file_in_place()`.

### PR DESCRIPTION
**Pull request by bryanforbes**
Fixes #2167

### CODE REVIEW

File: tests/test_black.py [hunk 1]
```diff
@@ -1578,7 +1578,33 @@ def test_reformat_one_with_stdin_filename(self) -> None:
                 mode=DEFAULT_MODE,
                 report=report,
             )
-            fsts.assert_called_once()
+            fsts.assert_called_once_with(
+                fast=True, write_back=black.WriteBack.YES, mode=DEFAULT_MODE
+            )
+            # __BLACK_STDIN_FILENAME__ should have been striped
```
Comment (JelleZijlstra) [inline tests/test_black.py:1584]:
```suggestion
            # __BLACK_STDIN_FILENAME__ should have been stripped
```

Comment (bryanforbes) [reply tests/test_black.py]: I fixed the two other instances in this file as well

File: CHANGES.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 ## Change Log
 
+### 21.4b3
```
Comment (JelleZijlstra) [inline CHANGES.md:3]:
```suggestion
### Unreleased
```

Let's not get ahead of ourselves :)

Comment (bryanforbes) [reply CHANGES.md:3]: Fair enough. I wasn't sure what to put for the version, so I guessed 😄

**Review by JelleZijlstra — APPROVED** [review]

**Review by ichard26 — APPROVED** [review]
LTGM, thank you for reporting _and fixing the bug!_ We really appreciate it!

**Pull request merged**